src/nmodl/nocpout.c

```diff
@@ -362,826 +380,912 @@ fprintf(stderr, "Notice: ARTIFICIAL_CELL models that would require thread specif
+            } else {
+                Sprintf(buf, "extern double %s;\n", s->name);
+            }
+            Lappendstr(defs_list, buf);
+        }
+    }
 
-	Lappendstr(defs_list, "static int _mechtype;\n\
+    Lappendstr(defs_list, "/* declaration of user functions */\n");
+    SYMLISTITER {
+        s = SYM(q);
+        if (s->subtype & (FUNCT | PROCED) && s->name[0] != '_') {
+            if (point_process) {
+                Sprintf(buf, "static double _hoc_%s();\n", s->name);
+            } else {
+                Sprintf(buf, "static void _hoc_%s(void);\n", s->name);
+            }
+            Lappendstr(defs_list, buf);
+        }
+    }
+
+    Lappendstr(defs_list,
+               "static int _mechtype;\n\
 extern void _nrn_cacheloop_reg(int, int);\n\
 extern void hoc_register_prop_size(int, int, int);\n\
 extern void hoc_register_limits(int, HocParmLimits*);\n\
 extern void hoc_register_units(int, HocParmUnits*);\n\
 extern void nrn_promote(Prop*, int, int);\n\
 extern Memb_func* memb_func;\n\
-"	);
-
-	if (nmodl_text) {
-		Lappendstr(defs_list,"\n"
-"#define NMODL_TEXT 1\n"
-"#if NMODL_TEXT\n"
-"static const char* nmodl_file_text;\n"
-"static const char* nmodl_filename;\n"
-"extern void hoc_reg_nmodl_text(int, const char*);\n"
-"extern void hoc_reg_nmodl_filename(int, const char*);\n"
-"#endif\n\n"
-		);
-	}
+");
 
-	/**** create special point process functions */
-	if (point_process) {
-		Lappendstr(defs_list, "extern Prop* nrn_point_prop_;\n");
-		Lappendstr(defs_list, "static int _pointtype;\n");
-		Lappendstr(defs_list, "static void* _hoc_create_pnt(_ho) Object* _ho; { void* create_point_process();\n");
-		Lappendstr(defs_list, "return create_point_process(_pointtype, _ho);\n}\n");
-		Lappendstr(defs_list, "static void _hoc_destroy_pnt();\n");
-		Lappendstr(defs_list, "static double _hoc_loc_pnt(_vptr) void* _vptr; {double loc_point_process();\n");
-		Lappendstr(defs_list, "return loc_point_process(_pointtype, _vptr);\n}\n");
-		Lappendstr(defs_list, "static double _hoc_has_loc(_vptr) void* _vptr; {double has_loc_point();\n");
-		Lappendstr(defs_list, "return has_loc_point(_vptr);\n}\n");
-		Lappendstr(defs_list, "static double _hoc_get_loc_pnt(_vptr)void* _vptr; {\n");
-		Lappendstr(defs_list, "double get_loc_point_process(); return (get_loc_point_process(_vptr));\n}\n");
-	}
-	/* function to set up _p and _ppvar */
-	Lappendstr(defs_list, "extern void _nrn_setdata_reg(int, void(*)(Prop*));\n");
-	Lappendstr(defs_list, "static void _setdata(Prop* _prop) {\n");
+    if (nmodl_text) {
+        Lappendstr(defs_list,
+                   "\n"
+                   "#define NMODL_TEXT 1\n"
+                   "#if NMODL_TEXT\n"
+                   "static const char* nmodl_file_text;\n"
+                   "static const char* nmodl_filename;\n"
+                   "extern void hoc_reg_nmodl_text(int, const char*);\n"
+                   "extern void hoc_reg_nmodl_filename(int, const char*);\n"
+                   "#endif\n\n");
```

Comment:
I like the way it handled this list of "...". I should convert all the multiline strings to this concatenated form.